tetris.py

```diff
@@ -370,4 +312,38 @@ def full_line_clear(self):
                     ] + [(a, b + 1) for (a, b) in coord_list if b < x_line]
 
 
+
+class TetrisGUI:
+    def __init__(self, speed, main_board):
+        self.speed = speed
+        self.rect_size = 25
+        self.main_board = main_board
+        self.start_time = time.time()
+        self.move_block()
+
+    def move_block(self):
+        """
+        Has the responsibility to call current_block_mover() and draw_block() to
+        simulate the blocks moving downwards on the canvas
+        """
+        self.main_board.current_block_mover()
```

Comment:
If `current_block_mover()` returns `True`, we make a new block:

```suggestion
        if self.main_board.current_block_mover():
            self.new_block()
```

You will need to add `new_block()` method.